models/data_ingestion.go

```diff
@@ -465,21 +463,23 @@ func (mb *MetricsBucket) Save(agentMsg *qanpb.CollectRequest) error {
 		return errors.New("Nothing to save - no metrics buckets")
 	}
 
-	// TODO: find solution with better performance
-	mb.db.Mapper = reflectx.NewMapperTagFunc("json", strings.ToUpper, func(value string) string {
-		if strings.Contains(value, ",") {
-			return strings.Split(value, ",")[0]
-		}
-		return value
-	})
 	tx, err := mb.db.Beginx()
 	if err != nil {
 		return fmt.Errorf("begin transaction: %s", err.Error())
 	}
+	var committed bool
+	defer func() {
+		if !committed {
+			tx.Rollback() //nolint:errcheck
```

Comment:
We **have to** commit or rollback transactions. Otherwise, database connection is leaked.